server/JsonStreamingServer/suppliers/JsonStreamingServer.Suppliers.FileStream/Services/CsvHotelReaderService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using JsonStreamingServer.Suppliers.FileStream.Services.Interfaces;
+using JsonStreamingServer.Core.Models.Domain;
+using CsvHelper;
+using Serilog;
+using System.Globalization;
+using System.Reflection;
+
+namespace JsonStreamingServer.Suppliers.FileStream.Services
+{
+    public class CsvHotelReaderService : ICsvHotelReaderService
+    {
+        private readonly ILogger _logger;
+        private int _index = 0;
+        private readonly List<HotelOffer> HotelOffersData = new();
+
+        public CsvHotelReaderService(ILogger logger)
```

Comment:
Please change the constructor. Add a file path as a parameter.

Reply:
I would create a new `StreamReader` based on the file stream and dispose of it inside IDisposable.